src/Mvc/Mvc.ViewFeatures/src/DefaultHtmlGenerator.cs

```diff
@@ -1329,27 +1339,15 @@ internal static object GetModelStateValue(ViewContext viewContext, string key, T
 
             case InputType.Text:
             default:
-                var attributeValue = (string)GetModelStateValue(viewContext, fullName, typeof(string));
-                if (attributeValue == null)
-                {
-                    attributeValue = useViewData ? EvalString(viewContext, expression, format) : valueParameter;
-                }
```

Comment:
Moving this around feels like an unrelated optimization. Am I missing something❔

Reply:
Yes, it's unrelated. I originally changed this because an earlier prototype modified this part of the code, and the duplicated logic/state was getting in the way. Happy to change it back if there's any concern with the change.